### PR TITLE
Moved the Model namespace into ORM so it can be distributed easier

### PR DESCRIPTION
**Pull request by lorenzo**
Also moved ModelAwareTrait to Datasource it I think it makes more sense there

### CODE REVIEW

**Comment by ADmad** [comment]
There seems to be one relevant failing test.

**Comment by ADmad** [comment]
I like this :+1:

File: src/ORM/BehaviorRegistry.php [hunk 1]
```diff
@@ -76,7 +76,11 @@ public function __construct(Table $table)
      */
     protected function _resolveClassName($class)
     {
-        return App::className($class, 'Model/Behavior', 'Behavior');
+        $result = App::className($class, 'Model/Behavior', 'Behavior');
+        if (!$result && in_array($class, ['CounterCache', 'Timestamp', 'Tree', 'Translate'])) {
+            return 'Cake\ORM\Behavior\\' . $class . 'Behavior';
```
Comment (markstory) [inline src/ORM/BehaviorRegistry.php]: This is going to be a pain to maintain in the future. Wouldn't another call to App:classname() also work here?

**Comment by markstory** [comment]
This will cause breaks if any userland behaviors extend core ones, but I think that should be relatively rare.

**Comment by lorenzo** [comment]
@markstory and also trivial to fix

**Comment by dereuromark** [comment]
I think behaviors should stay the way they are.
It will be more confusing to have the core ones in a different namespace than the user ones, this is not the case for any other type of class.

**Comment by ADmad** [comment]
I thought App\Model would also be changed to App\ORM.

**Comment by dereuromark** [comment]
Ah, well then it makes sense :)

**Comment by markstory** [comment]
@ADmad I wouldn't expect the app namespaces to change at all because of this.

**Comment by ADmad** [comment]
@markstory In that case i share the same concern as @dereuromark. Having core behaviors and user behaviors under different namespaces would be bad and confusing for users imo too.

**Comment by lorenzo** [comment]
@ADmad Why is it confusing? The structure of an app does not have to be a mirror of how the framework is structured.

**Comment by lorenzo** [comment]
Or in this case it would be the opposite. The only reason it was put into `Model` was because `App::classname` was forcing us to put it there.

**Comment by dereuromark** [comment]
```php
use Cake\ORM\Behavior\TimestampBehavior
```
but
```php
use App\Model\Behavior\TimestampBehavior
use PluginjName\Model\Behavior\TimestampBehavior
```

I find that rather confusing somehow.

**Comment by ADmad** [comment]
@lorenzo We have users who get confused by even lack of `<?php` tags in code examples of manual, hence my concern :)

**Comment by lorenzo** [comment]
@ADmad well, I get that those users are not looking forward contributing to the core, so what is the concern exactly?

**Comment by lorenzo** [comment]
@dereuromark where do you directly "use" a behavior?

**Comment by dereuromark** [comment]
That was just one example. Maybe when extending it.
But in general, the different namespaces might not be clear/straightforward to most.

**Comment by lorenzo** [comment]
@dereuromark I guess it is supper confusing that you extend `Cake\ORM\Table` then

File: src/ORM/BehaviorRegistry.php (hunk 1 above)
Comment (lorenzo) [inline src/ORM/BehaviorRegistry.php]: Yeah, I can change it to use that it will be marginally slower, though.

**Comment by lorenzo** [comment]
Used another call to `App::classname` to avoid the hardcoding.

**Comment by markstory** [comment]
Looks good :+1:

**Pull request merged**